sycl/include/sycl/ext/oneapi/experimental/annotated_ptr/annotated_ptr.hpp

```diff
@@ -120,23 +120,23 @@ class annotated_ref<T, detail::properties_t<Props...>> {
 
   template <class O, class P>
   T operator=(const annotated_ref<O, P> &Ref) const {
-    O t2 = Ref;
+    O t2 = Ref.operator T();
     return *this = t2;
```

Comment:
Style question: Does someone care whether this is instead
```suggestion
    O t2 = *(Ref.m_Ptr);
    return *this = t2;
```
or
```suggestion
    return *this = *(Ref.m_Ptr);;
```
or
```suggestion
    return *this = Ref.operator T();
```

Reply:
O t2 = *(Ref.m_Ptr); // this is wrong, we need to conversion to have annotation
return *this = *(Ref.m_Ptr);; // same

 return *this = Ref.operator T(); // this is ok, I have no preference

Reply:
you're right. If someone prefers we avoid the `.operator` syntax we could move the annotated load into a private method in change this to 
```suggestion
    return *this = Ref.load();
```

It might even make sense to add this to the public interface (and add it to the spec). Because user code could have the same problem and appreciate not to have to use `.operator` in that case.

Reply:
I will keep the .operator for now

Reply:
@Pennycook @gmlueck Do you think we should add load (or different name like get, ...) so users don't have the same problem if they have a type with conflicting conversion operator? If so we can do that in a separate PR.

Reply:
Honestly, I'm not even sure I understand what these few lines are supposed to do.  Could you give an example of where somebody might need to use this, outside of the definition of this operator?

Reply:
```
void f(annotated_ptr<MyT, ....> p) {
   MyT v = *p;
}
```
This works for any reasonable type. The `operator*` returns an `annotated_ref` and the assignment triggers the implicit conversion which does the annotated load. The problem is if the type is badly written and has an un/under-constrained conversion constructor e.g.:
```
struct MyT {
    template<class T> MyT(T&&) { ...}
};
```
Now that gets used rather than the implicit conversion from annotated_ref which is the intention. If a user has to deal with this kind of type (or wants to program generically very defensively that any such weird type would work) they then would instead write:
```
void f(annotated_ptr<MyT, ....> p) {
   MyT v = p->load();
}
```
Given that this doesn't rely on implicit conversion, there is no way for the type to interfere.

Reply:
I see.  Thanks for the explanation.  I agree that adding a member function would improve usability.

My preference would be to call it `load()`, because that's aligned with `atomic_ref` (which presumably has the same problem, since it also exposes an implicit conversion to `T`?).  For symmetry and consistency we should define `store(T)` as well.